developer-docs/bolt_server.md

```diff
@@ -0,0 +1,60 @@
+# Bolt Server 
+
+## Overview
+Bolt server provides a service to run bolt tasks over SSH and WinRM. This service is exposed via an API as described in this document. 
+
+## Configuration
+Bolt server can be configured by defining content in HOCON format at one of the following expected configuration file path locations.
+
+**Global Config**: `/etc/puppetlabs/bolt-server/conf.d/bolt-server.conf`
+
+**Local Config**: `~/.puppetlabs/bolt-server.conf`
+
+**Options**
+- `port`: Integer, *optional* - The port the bolt server will run on (default 8144)
+- `ssl-cert`: String, *required* - Path to the cert file. **Note** PE default: `/etc/puppetlabs/bolt-server/$HOSTNAME.cert.pem`
+- `ssl-key`: String, *required* - Path to the key file
+- `ssl-ca-cert`: String, *required* - Path to the ca cert file
+
+**Example**
+```
+bolt-server: {
+    port: 8144
+    ssl-cert: /etc/puppetlabs/bolt-server/ssl/cert.pem
+    ssl-key: /etc/puppetlabs/bolt-server/ssl/private_key.pem
+    ssl-ca-cert: /etc/puppetlabs/bolt-server/ssl/ca.pem
+} 
+```
+
+## API Endpoints
+Each API endpoint accepts a request as described below. The request body must be a JSON object.
+
+## POST /ssh/run_task
+- `target`: [SSH Target Object](#SSH Target Object), *required* - Target information to run task on.
+- `task`: [Task Object](#Task Object), *required* - Task to run on target.
+- `parameters`: Object, *optional* - JSON formatted parameters to be provided to task.
+
+## SSH Target Object
+The Target be a JSON object. The following keys are available:
+- `hostname`: String, *required* - Target identifier.
+- `password`: String, *required unless `private-key` is specified* - Password for SSH transport authentication.
+- `private-key`: String, *required unless `password` is specified* - Key for ssh transport authentication.
```

Comment:
I think we should remove this until it's implemented. Most keys get passed directly to bolt, but we have to add some handling for this one :/